docs/api/app.md

```diff
@@ -309,6 +309,39 @@ app.on('certificate-error', (event, webContents, url, error, certificate, callba
 })
 ```
 
+### Event: 'client-certificate-request-password' _Linux_
+
+Returns:
+
+* `event` Event
+* `hostName` string - the hostname of the site requiring a client certificate
+* `tokenName` string - the token (or slot) name of the cryptographic device
+* `retry` boolean - whether there have been failed attempt at prompting the password
```

Comment:
Positional parameters for events should be avoided in new events, see https://github.com/electron/governance/blob/main/wg-api/best-practices.md#how-will-this-api-be-extended-in-the-future.

Best would be to have these be properties of the emitted `Event` object. This is a relatively new style in Electron so there aren't too many examples, but we are planning to refactor more events to use this style. See https://github.com/electron/electron/blob/e67ab9a93dadccecff30de50ab4555191c30b6c4/shell/browser/api/electron_api_web_contents.cc#L1912-L1924 for one example of how to do this.

Reply:
I have made the change. I have not been able to attach the callback the dictionary. The build was failing with 
```
../../gin/converter.h:325:38: error: call to deleted constructor of 'base::OnceCallback<void (Arguments *)>'
  325 |   return Converter<T>::ToV8(isolate, input);                                                                                     
```

So callback is still passed as an  extra argument. But I guess it's fine since it does not appear in the public API. Also, this allows me to reuse `ClientCertRequestParams `type as is an `internal-electron.d.ts`